### PR TITLE
[Bug Fix] LPS-152675

### PR DESCRIPTION
**Pull request by JanisZhang**
Hi @dantewang @tinatian,

This is for https://issues.liferay.com/browse/LPS-152675 , a regresstion bug caused by HttpComponentsUtil .Thanks !

FYI @vicnate5 

Janis.

### CODE REVIEW

File: portal-kernel/src/com/liferay/portal/kernel/util/HttpComponentsUtil.java [hunk 1]
```diff
@@ -252,7 +252,9 @@ public static String getDomain(String url) {
 			uri = getURI(url);
 		}
 		catch (URISyntaxException uriSyntaxException) {
-			_log.error(uriSyntaxException);
+			if (_log.isDebugEnabled()) {
```
Comment (dantewang) [inline portal-kernel/src/com/liferay/portal/kernel/util/HttpComponentsUtil.java:255]: @JanisZhang The original `HttpImpl` has the try-catch-log-debug block in the `getURI(String)` method. When we moved this logic, the method was changed to directly throw the exception. This causes logic changes not only in the methods/logic modified in this PR.

For example, in `LayoutReferenceExportImportContentProcessor#_getPortalURL`,

```
	private String _getPortalURL(String url, String portalURL)
		throws PortalException {

		try {
			URI uri = HttpComponentsUtil.getURI(url);

			if ((uri != null) &&
				InetAddressUtil.isLocalInetAddress(
					InetAddress.getByName(uri.getHost()))) {

				return StringBundler.concat(
					uri.getScheme(), "://", uri.getHost(), StringPool.COLON,
					uri.getPort());
			}
		}
		catch (UnknownHostException unknownHostException) {
			if (_log.isDebugEnabled()) {
				_log.debug(unknownHostException);
			}
			else if (_log.isWarnEnabled()) {
				_log.warn(unknownHostException);
			}
		}
		catch (Exception exception) {
			throw new PortalException(exception);
		}

		return portalURL;
	}
```

The logic is to do a null check upon the return value, and return the original string if the `uri` is null. After the move, `getURI` throws an exception, so now for an illegal `url` it goes to the `catch (Exception exception)` block, which breaks the original logic of this method.

If we keep the current logic of `getURI`, please also update this usage, and also the logging in `PortalImpl#escapeRedirect`.

Or just add the try-catch-log-debug block back to `getURI` to restore its original contract.

**Review by dantewang — CHANGES_REQUESTED** [review]

File: portal-kernel/src/com/liferay/portal/kernel/util/HttpComponentsUtil.java (hunk 1 above)
Comment (JanisZhang) [reply portal-kernel/src/com/liferay/portal/kernel/util/HttpComponentsUtil.java:255]: @dantewang Got it

**Comment by JanisZhang** [comment]
ci:stop

**Comment by JanisZhang** [comment]
ci:test:sf

**Comment by JanisZhang** [comment]
ci:test:relevant

**Comment by JanisZhang** [comment]
ci:test:acceptance

**Comment by JanisZhang** [comment]
Hi @catarineaguiar

Could you please help to scan the test results, THX!

**Comment by catarineaguiar** [comment]
ci:test:sf

**Comment by tinatian** [comment]
@JanisZhang Your change is going to swallow the exceptions which is dangerous most time, we should check the staging case and see if it really needs to swallow the exceptions, then fix it on staging code rather than the API code.

FYI @dantewang 

Tina.

**Comment by JanisZhang** [comment]
Hi @tinatian ,

The original `HttpImpl` is to add try-catch-log-debug in `getURI()` , so I changed to swallow the exceptions on staging code which uses `HttpComponentUtil.getURI()` . 

For methods `getDomain(String url) `, `getPath(String url)`, `getProtocol(String url)`, `getQueryString(String url)`  in HttpComponent, there are just try to get a  component from URL.

For **LayoutReferencesExportImportContentProcessor**#`_getPortalURL` , **PortalImpl**#`escapeRedirect`
It looks ok to pass in an invalid URI.  For the staging case, it obviously passes a string in JSON format. Thanks.

cc @dantewang

**Comment by tinatian** [comment]
@JanisZhang Could you please change the commit message to "change log level back to debug to make sure it consistent with the behavior before extracting from HttpImpl (LPS-150185 ad67ecb4e0471ea67fb1d130664037605b31194c)"

So the reason you make this change is to make sure it consistent with before, not that you decided to lower down the log level for staging to avoid seeing errors on log. Swallowing exception needs more strong reason to do than avoid seeing logs.

Tina.

**Comment by JanisZhang** [comment]
Hi @tinatian, Ok. Yes. Thanks!

**Comment by JanisZhang** [comment]
ci:test:sf

**Comment by tinatian** [comment]
@shuayngzhou This pull is good to go.

Tina.

**Comment by shuyangzhou** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/117384

**Comment by shuyangzhou** [comment]
ci:close

**Pull request closed without merging**